includes/discovery/ports.inc.php

```diff
@@ -44,9 +44,25 @@
 foreach ($port_stats as $ifIndex => $port) {
     // Store ifIndex in port entry and prefetch ifName as we'll need it multiple times
     $port['ifIndex'] = $ifIndex;
-    // handle ifName and ifAlias the same way we do in the poller
-    $ifName = empty($port['ifName']) ? $port['ifDescr'] : $port['ifName'];
-    $ifAlias = empty($port['ifAlias']) ? $port['ifDescr'] : $port['ifAlias'];
+
+    // When devices do not provide ifDescr data, populate with ifName data if available
```

Comment:
I've added the 'is_port_valid' function before doing the copy actions so no new ports are added. The travis test still fails but as far as i can see this is "correct" because we are now updating the fields that previously were empty and were filed by the poller. 
Discovery is now filling them the same way the poller does.